packages/eslint-plugin/docs/rules/no-floating-promises.mdx

```diff
@@ -115,6 +115,66 @@ await (async function () {
 })();
 ```
 
+### `allowForKnownSafePromises`
+
+This allows you to skip checking of those promises which are marked with some kind of branded type. The branding of a type can be done either by using a type alias or an interface.
+
+Each item must be one of :
+
+- A type defined in a file (`{from: "file", name: "Foo", path: "src/foo-file.ts"}` with `path` being an optional path relative to the project root directory)
+- A type from the default library (`{from: "lib", name: "PromiseLike"}`)
+- A type from a package (`{from: "package", name: "Foo", package: "foo-lib"}`, this also works for types defined in a typings package).
+
+Additionally, a type may be defined just as a simple string, which then matches the type independently of its origin.
```

Comment:
[Docs] Let's, ah, not mention this. We really don't want people to use it. See linked issues' discussions around naming: catch-all names tend to be overly permissive.

Eventually/separately we'll set up unified docs for the format and mention the preference against strings.

Reply:
I don't think I understood this.
The linked issue says this right ?
> I propose we adopt the format discussed in https://github.com/typescript-eslint/typescript-eslint/discussions/6017 & implemented in https://github.com/typescript-eslint/typescript-eslint/pull/4436 ([see v6 docs](https://v6--typescript-eslint.netlify.app/rules/prefer-readonly-parameter-types/#allow)) as a new `allow` option for `no-floating-promises`.

Reply:
Yup, and none of us on the team are enthused that the `string` type exists. At the time we'd added it for compatibility with the existing rule. Not because we particularly wanted it to exist.

Especially for this new rule, I don't think we want users to really be encouraged to use it.